internal/api/server.go

```diff
@@ -232,11 +259,71 @@ func (s *Server) ReshareVault(c echo.Context) error {
 		asynq.Retention(10*time.Minute),
 		asynq.Queue(tasks.QUEUE_NAME))
 	if err != nil {
+		s.logger.Errorf("ReshareVault: Failed to enqueue task: %v", err)
 		return fmt.Errorf("fail to enqueue task, err: %w", err)
 	}
+
+	// Notify plugin server asynchronously
+	go func() {
+		defer func() {
+			if r := recover(); r != nil {
+				s.logger.Errorf("ReshareVault: Panic in async plugin notification: %v", r)
+			}
+		}()
+
+		bgCtx, cancel := context.WithTimeout(context.Background(), 30*time.Second)
+		defer cancel()
+
+		if err := s.notifyPluginServerReshare(bgCtx, req); err != nil {
+			s.logger.Errorf("ReshareVault: Failed to notify plugin server: %v", err)
+		}
+	}()
+
 	return c.NoContent(http.StatusOK)
 }
 
+// notifyPluginServerReshare sends the reshare request to the plugin server
+func (s *Server) notifyPluginServerReshare(ctx context.Context, req tv.ReshareRequest) error {
```

Comment:
Do this from the front end

Reply:
but should we expose this function for UI to use or directly call the plugin server from FE?